spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/ribbon/apache/RibbonLoadBalancingHttpClientTests.java

```diff
@@ -247,7 +247,7 @@ public void testRetrySameServerOnly() throws Exception {
 		doReturn(uriRequest).when(request).toRequest(any(RequestConfig.class));
 		RibbonApacheHttpResponse returnedResponse = client.execute(request, null);
 		verify(delegate, times(2)).execute(any(HttpUriRequest.class));
-		verify(lb, times(0)).chooseServer(eq(serviceName));
+		verify(lb, times(1)).chooseServer(eq(serviceName));
```

Comment:
I'm a little concerned that the number of times `chooseServer()` is called has increased. We've seen issues where, when the number of servers is small (2), one is always skipped since choose was called multiple times. Can you explain why this changed (and for the 4 below as well)?

Reply:
Actually the number of servers chosen by the LoadBalancer is the same.  Prior to this change the first server instance was chosen by the LoadBalancerCommand.   Now that the AbstractLoadBalancerAwareClient is invoking the execute method for retriable requests the first server is chosen by the InterceptorRetryPolicy.

This can be verified by resetting the RetryPolicy for both the RetryableRibbonLoadBalancingHttpClient and RetryableOkHttpLoadBalancingClient back to the FeignRetryPolicy.  In that scenario the first request will fail because a server has not been chosen and the host and port on the request are undefined.